specs/node_types.md

```diff
@@ -0,0 +1,105 @@
+
+#### Full Bodies
+
+These nodes fully download and validate [the erasure coding](./data_structures.md#2d-reed-solomon-encoding-scheme) of all block bodies.
+
+If [transactions are not processed](#no-transactions), secure under an honest minority of nodes and a weak subjectivity assumption. If [transactions are processed](#full-transactions), secure under a weak subjectivity assumption.
+
+### Transactions
+
+#### No Transactions
+
+These nodes process do not process requests [with a reserved namespace ID](./data_structures.md#arranging-available-data-into-shares) and thus to not know the chain state without relying on a third party.
+
+At most secure under an honest minority of nodes and a weak subjectivity assumption.
+
+#### Full Transactions
+
+Nodes that wish to produce new blocks must know the [chain state](./data_structures.md#state). Processing all block bodies is actually not needed to know the LazyLedger state, as [transactions that pay for message inclusion commit to messages](./../rationale/message_block_layout.md). These nodes process all requests [with a reserved namespace ID](./data_structures.md#arranging-available-data-into-shares) from block bodies and perform Data Availability Sampling for the remaining (message) data.
+
+At most secure under a weak subjectivity assumption.
+
+## Node Types
+
+For convenience, we will define several common parameter configurations:
+1. [Full nodes](https://en.bitcoin.it/wiki/Full_node) provide the strongest security guarantees. Block bodies are not stored.
+    - Block headers: [Extended Block Headers](#extended-block-headers)
+    - Block bodies: [Full Bodies](#full-bodies)
+    - Transactions: [Full Transactions](#full-transactions)
+1. Partial nodes are capable of producing fraud proofs of invalid transactions and contribute to validating the erasure coding of random blocks.
+    - Block headers: [Extended Block Headers](#extended-block-headers)
+    - Block bodies: [Partial Bodies](#partial-bodies)
+    - Transactions: [Full Transactions](#full-transactions)
+1. Light nodes DAS and are secure under an honest minority.
+    - Block headers: [Extended Block Headers](#extended-block-headers)
+    - Block bodies: [Sampled Bodies](#sampled-bodies)
+    - Transactions: [No Transactions](#no-transactions)
+1. Superlight nodes do not perform DAS and are secure under an honest majority.
+    - Block headers: [Compact Block Headers](#compact-block-headers)
+    - Block bodies: [No Bodies](#no-bodies)
+    - Transactions: [No Transactions](#no-transactions)
+1. Light validator nodes can produce new blocks with strong security guarantees and light resource requirements.
+    - Block headers: [Extended Block Headers](#extended-block-headers)
+    - Block bodies: [Sampled Bodies](#sampled-bodies)
+    - Transactions: [Full Transactions](#full-transactions)
+1. Storage nodes provide the same security guarantees as full nodes. Block bodies are stored and served to the network.
```

Comment:
Would you make a difference between (1) a node that stores and serves the raw availableData and (2) a node that stores and serves the erasure-coded version of that data (e.g. an ipfs node that serves the rows and columns)? Or would you rather say, it's the same thing and just a different encoding? 

The interfaces to interact with these nodes is quite different though:
 - (1) give me block data of block with hash x / height x
 - (2) give me leaf-data of the erasure-block with hash x with index y (or path, cid/0/0/1/0/0), or, give me all leaves of namespace nid (could also be a path root/nid)

Reply:
I don't really see a need to distinguish the two cases. It's two different encodings of the same data.

But technically, when placed in a 2D square, the available data actually gains information in the form of specific placement of shares that isn't captured with the raw available data (since the block proposer can include padding shares). As such, storage nodes should probably be storing and serving the erasure-coded version of the block bodies.

Reply:
Should we then change this to
```suggestion
1. Storage nodes provide the same security guarantees as full nodes. Block bodies (in [erasure-coded form](./data_structures.md#2d-reed-solomon-encoding-scheme)) are stored and served to the network.
```
?